sweeper/aws_route53_resolver_firewall_rule_group_association: failing with `ValidationException`

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Actual Behavior
```
error deleting Route 53 Resolver DNS Firewall rule group association (rslvr-frgassoc-37b8b578ef654038): ValidationException: [RSLVR-02307] The DNS Firewall rule group association is under mutation protection. Remove the mutation protection and try again. Trace Id: "1-616663a9-5103cdcf2248c4993a1e43a7"
```

```
% TEST=./aws SWEEP=us-west-2 SWEEPARGS=-sweep-run=aws_route53_resolver_firewall_rule_group make sweep
WARNING: This will destroy infrastructure. Use only in development accounts.
go test ./aws -v -sweep=us-west-2 -sweep-run=aws_route53_resolver_firewall_rule_group -timeout 60m
go: downloading github.com/aws/aws-sdk-go v1.41.1
2021/10/13 12:54:14 [WARN] Sweeper has dependency (aws_route53_resolver_firewall_rule_association), but that sweeper was not found
2021/10/13 12:54:14 [DEBUG] Running Sweepers for region (us-west-2):
2021/10/13 12:54:14 [DEBUG] Sweeper (aws_route53_resolver_firewall_rule_group) has dependency (aws_route53_resolver_firewall_rule), running..
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	3.326s
FAIL
make: *** [sweep] Error 1
```

--- a/aws/resource_aws_route53_resolver_firewall_rule_group_association_test.go
+++ b/aws/resource_aws_route53_resolver_firewall_rule_group_association_test.go
@@ -12,6 +12,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/route53resolver/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/route53resolver/waiter"
 )
 
 func init() {
@@ -41,6 +42,31 @@ func testSweepRoute53ResolverFirewallRuleGroupAssociations(region string) error 
 			r := resourceAwsRoute53ResolverFirewallRuleGroupAssociation()
 			d := r.Data(nil)
 			d.SetId(id)
+
+			if aws.StringValue(firewallRuleGroupAssociation.MutationProtection) == route53resolver.MutationProtectionStatusEnabled {
+				input := &route53resolver.UpdateFirewallRuleGroupAssociationInput{
+					FirewallRuleGroupAssociationId: firewallRuleGroupAssociation.Id,
+					Name:                           firewallRuleGroupAssociation.Name,
+					MutationProtection:             aws.String(route53resolver.MutationProtectionStatusDisabled),
+				}
+
+				_, err := conn.UpdateFirewallRuleGroupAssociation(input)
+
+				if err != nil {
+					log.Printf("[ERROR] %s", err)
+					sweeperErrs = multierror.Append(sweeperErrs, err)
+					continue
+				}
+
+				_, err = waiter.FirewallRuleGroupAssociationUpdated(conn, d.Id())
+
+				if err != nil {
+					log.Printf("[ERROR] error waiting for Route53 Resolver DNS Firewall rule group association (%s) to be updated: %s", d.Id(), err)
+					sweeperErrs = multierror.Append(sweeperErrs, err)
+					continue
+				}
+			}
+
 			err := r.Delete(d, client)
 
 			if err != nil {

--- a/aws/resource_aws_route53_resolver_firewall_rule_test.go
+++ b/aws/resource_aws_route53_resolver_firewall_rule_test.go
@@ -19,7 +19,7 @@ func init() {
 		Name: "aws_route53_resolver_firewall_rule",
 		F:    testSweepRoute53ResolverFirewallRules,
 		Dependencies: []string{
-			"aws_route53_resolver_firewall_rule_association",
+			"aws_route53_resolver_firewall_rule_group_association",
 		},
 	})
 }

